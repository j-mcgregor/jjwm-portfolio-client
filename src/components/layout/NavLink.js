import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const NavLink = ({ page, title, closeModal }) => {
  const { pathname } = window.location;

  return (
    <li className="nav-item">
      <Link
        to={`/${page}`}
        className={`nav-link ${pathname === `/${page}` ? 'text-muted' : ''}`}
        onClick={closeModal}
      >
        {title}
      </Link>
    </li>
  );
};

NavLink.propTypes = {
  page: PropTypes.string,
  title: PropTypes.string,
  closeModal: PropTypes.func
};

NavLink.defaultProps = {
  page: '',
  title: '',
  closeModal: () => {}
};

export default NavLink;
