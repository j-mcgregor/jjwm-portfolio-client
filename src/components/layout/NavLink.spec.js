import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { shallow, mount } from 'enzyme';
import NavLink from './NavLink';

describe('NavLink', () => {
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      page: 'test',
      title: 'Test',
      closeModal: jest.fn()
    };
    Object.defineProperty(window, 'location', {
      value: {
        pathname: '/test'
      }
    });
  });

  afterEach(() => {});

  it('should render correctly', () => {
    const wrapper = shallow(<NavLink {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Link').text()).toBe('Test');
    expect(wrapper.find('Link').prop('to')).toBe('/test');
  });

  it('should trigger closeModal when clicked', () => {
    const wrapper = shallow(<NavLink {...defaultProps} />);
    wrapper.find('Link').simulate('click');
    expect(defaultProps.closeModal).toHaveBeenCalled();
  });

  it('should NOT have the "text-muted" class when the window location does not match the page', () => {
    window.location.pathname = '/blah';

    const wrapper = mount(
      <Router>
        <NavLink {...defaultProps} />
      </Router>
    );

    const isMuted = wrapper
      .find('Link')
      .prop('className')
      .includes('text-muted');

    expect(isMuted).toBe(false);
  });

  it('should have the "text-muted" class when the window location matches page', () => {
    window.location.pathname = '/test';

    const wrapper = mount(
      <Router>
        <NavLink {...defaultProps} />
      </Router>
    );

    const isMuted = wrapper
      .find('Link')
      .prop('className')
      .includes('text-muted');

    expect(isMuted).toBe(true);
  });
});
