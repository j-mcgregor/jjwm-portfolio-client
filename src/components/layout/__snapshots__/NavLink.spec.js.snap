// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink should render correctly 1`] = `ShallowWrapper {}`;
